server_src/runners/state.js

```diff
@@ -3409,29 +3416,21 @@ class Runner {
       if (settings.systemCommands.serverRestart !== "") {
         farmPrinters[index].settingsServer.commands.serverRestartCommand =
           settings.systemCommands.serverRestart;
-      } else {
-        settings.systemCommands.serverRestart =
-          farmPrinters[index].settingsServer.commands.serverRestartCommand;
       }
       if (settings.systemCommands.systemRestart !== "") {
         farmPrinters[index].settingsServer.commands.systemRestartCommand =
           settings.systemCommands.systemRestart;
-      } else {
-        settings.systemCommands.systemRestart =
-          farmPrinters[index].settingsServer.commands.systemRestartCommand;
       }
       if (settings.systemCommands.systemShutdown !== "") {
         farmPrinters[index].settingsServer.commands.systemShutdownCommand =
           settings.systemCommands.systemShutdown;
-      } else {
-        settings.systemCommands.systemShutdown =
-          farmPrinters[index].settingsServer.commands.systemShutdownCommand;
       }
 
       printer.save().catch((e) => {
         logger.error(JSON.stringify(e), "ERROR savin power settings.");
       });
-      if (settings.state !== "Offline") {
+      // Made the state check from the server, not the client...
+      if (farmPrinters[index].state !== "Offline") {
```

Comment:
This was causing the issue of the client not rescanning after a URL change. The client was sending "Offline" state so it was never triggering.